app/models/grda_warehouse_base.rb

```diff
@@ -31,4 +31,11 @@ def self.needs_migration?
   def self.partitioned?(table_name)
     Dba::PartitionMaker.new(table_name: table_name).done?
   end
+
+  # default colocated versions table for warehouse records
+  def self.has_paper_trail(options = {}) # rubocop:disable Naming/PredicateName
+    versions = options.fetch(:versions, {}).merge(class_name: 'GrdaWarehouse::Version')
```

Comment:
Does this let you override which versions table is used? We may want to do that for [GrdaWarehouse::UserViewableEntity](https://github.com/greenriver/hmis-warehouse/blob/0d11a228fcb9926a63f8e2e1987d1a6d9ccc67a1/app/models/grda_warehouse/user_viewable_entity.rb#L9-L14) to keep its versions in the app db, since it needs those metadata columns and references both dbs. But @eanders can confirm..

Reply:
Good catch, I think that since user_viewable_entities lives in the warehouse database, we should use the warehouse versions table to track it. I can add referenced_user_id and referenced_entity_name to the warehouse versions table as well

Reply:
Hmm, yeah, I agree with @ttoomey we should keep things in the DB the table exists in.  This change will require updating the UI for auditing users, and maybe clients.

Reply:
@eanders I was looking for potential places this would break the warehouse views. As far as I can tell, there are 2 places we show "audit history":
* Client dashboard "audit" tab (Access History) uses [Admin::AuditsController](https://github.com/greenriver/hmis-warehouse/blob/release-94/app/controllers/admin/audits_controller.rb) which is based on ActivityLog, so it should not be affected. The client audit just shows who has viewed the client, it does not show changes to the client record. Can you confirm?
* User audit history uses [Admin::EditHistoriesController](https://github.com/greenriver/hmis-warehouse/blob/release-94/app/controllers/admin/edit_histories_controller.rb), which I think should correctly be based off of the App DB since that's where Users and access control-related things live. But if UserViewableEntity versions are in the other DB, sounds like it would complicate that.

Reply:
For reference @ttoomey 
https://qa-hmis-warehouse.openpath.host/clients/==0g5mRw2yHRr7rkV7q1mKCw==/audits
https://qa-hmis-warehouse.openpath.host/admin/users/==kM3z30Nc2+STWyq9o+9f0g==/edit_history

Reply:
@gigxz , that matches my understanding, I'd forgotten about the client side using activity log (I think that's what the user view history uses too.  This should be fine for the warehouse, the number of places someone can change a client record are very small.) Looks like the [EditHistoriesController](https://github.com/greenriver/hmis-warehouse/blob/release-94/app/controllers/admin/edit_histories_controller.rb) would need an update to somehow union the changes from both versions tables if we move the user viewable entities.  That said, at least in one installation, it looks like we changed something so that these haven't seen a change since 2019.

Reply:
Ah yeah, I can't find any meaningful usages of `GrdaWarehouse::UserViewableEntity` in the code base.. just a rake task that was written 4 years ago. The user model doesn't have a relationship to it at all. So, that seems safe to ignore, and maybe we should remove it entirely.

Reply:
Hmm, maybe, pretty sure those transitioned to `GroupViewableEntity` which may have a similar issue.

Reply:
We don't have PaperTrail enabled on GroupViewableEntity 🧐 
Maybe a consideration for [this ticket](https://www.pivotaltracker.com/story/show/186497193).